specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json

```diff
@@ -0,0 +1,5736 @@
+          "type": "string"
+        },
+        "containerName": {
+          "description": "Gets or sets the container name to share.",
+          "type": "string"
+        },
+        "providerPath": {
+          "description": "Gets or sets the path to file/folder within the container.",
+          "type": "string"
+        }
+      }
+    },
+    "StorageAccountProperties": {
+      "description": "Properties of the Storage account data set.",
+      "required": [
+        "storageAccountResourceId",
+        "subscriptionId",
+        "resourceGroup",
+        "storageAccountName"
+      ],
+      "type": "object",
+      "properties": {
+        "dataSetId": {
+          "description": "Unique id for identifying a data set resource",
+          "type": "string",
+          "readOnly": true
+        },
+        "location": {
+          "description": "Location of the storage account.",
+          "type": "string"
+        },
+        "paths": {
+          "description": "A list of paths under a given storage account.",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/StorageAccountPath"
+          }
+        },
+        "resourceGroup": {
+          "description": "Resource group of storage account",
+          "type": "string"
+        },
+        "storageAccountName": {
+          "description": "Storage account name of the source data set",
+          "type": "string"
+        },
+        "storageAccountResourceId": {
+          "description": "Resource id of the storage account.",
+          "type": "string"
+        },
+        "subscriptionId": {
+          "description": "Subscription id of storage account",
+          "type": "string"
+        }
+      }
+    },
+    "SynapseSparkPipelineStep": {
+      "description": "A Synapse Spark based pipeline step.",
+      "required": [
+        "properties",
+        "kind"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/PipelineStep"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/SynapseSparkPipelineStepProperties",
+          "description": "Properties of Synapse Spark pipelineStep",
+          "x-ms-client-flatten": true
+        }
+      },
+      "x-ms-discriminator-value": "SynapseSparkPipelineStep"
+    },
+    "SynapseSparkPipelineStepProperties": {
+      "description": "A Synapse Spark PipelineStep data transfer object.",
+      "required": [
+        "clusterSize",
+        "scriptId",
+        "scriptRevision",
+        "scriptSourceBindings",
+        "scriptSinkBindings"
+      ],
+      "type": "object",
+      "properties": {
+        "clusterSize": {
+          "format": "int32",
+          "description": "Defines the cluster size for running a SynapseSpark pipeline step",
+          "type": "integer"
+        },
+        "dependencies": {
```

Comment:
Why is this part of just synapseSpark and not the generic pipelinestep contract?

Reply:
This is how polymorphic types are modeled in the entire file, base class is not really holding the base property class (in code that is) so the swagger is generated with all derived class holding all shared properties.
This is how it's model in other RPs as well (specifically in ADS, we use the same generator and patterns).

Would prefer to leave it like this, at least for where we stand now